core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -39,225 +44,251 @@
  * The syntax of the metrics configuration file is:
  * [instance].[sink|source].[name].[options]=[value]
  */
-@NotThreadSafe
-public class MetricsSystem {
+@ThreadSafe
+public final class MetricsSystem {
```

Comment:
I think we have trying to move away from singletons, can MetricsSystem only be implemented as a singleton?